code/js/modules/BaseController.js

```diff
@@ -168,6 +205,76 @@
     return isPlaying;
   };
 
+  BaseController.prototype.setPosition = function(time) {
+    if(this.selectors.video) {
+      this.doc().querySelector(this.selectors.video).currentTime = time;
+    }
+  };
+
+  BaseController.prototype.seek = function(time) {
+    // default implementation uses selectors if present
+    if(this.selectors.video) {
+      this.doc().querySelector(this.selectors.video).currentTime += time;
+    }
+  };
+
+  BaseController.prototype.getCurrentTime = function() {
+    // default implementation uses selectors if present
+    if(this.selectors.currentTime) {
+      var timestr = (this.getSongData(this.selectors.currentTime) || "").trim();
```

Comment:
Just found out that this is incorrect. It should be `|| "0"` instead of `|| ""`.
I will send a patch to fix that in the current version anyway, I just wanted to mention that here too, in case this gets merged soon.

Reply:
Thank you for your review, but I have mostly given up working on streamkeys. Instead I started from fresh